lib/opFns.js

```diff
@@ -260,10 +258,16 @@ module.exports = {
     return runState.callValue
   },
   CALLDATALOAD: function (pos, runState) {
-    pos = utils.bufferToInt(pos)
-    var loaded = runState.callData.slice(pos, pos + 32)
+    pos = new BN(pos)
 
-    loaded = loaded.length ? loaded : Buffer.from([0])
+    var loaded
+    if (pos > runState.callData.length) {
+      loaded = Buffer.from([0])
+    } else {
+      pos = pos.toNumber()
+      loaded = runState.callData.slice(pos, pos + 32)
+      loaded = loaded.length ? loaded : Buffer.from([0])
+    }
 
```

Comment:
Hmmm, to just change this would have been the cleaner solution in my regard, but ok.